src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.azure.cosmos.cassandra;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.connection.ClosedConnectionException;
+import com.datastax.oss.driver.api.core.connection.HeartbeatException;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.servererrors.WriteType;
+import com.datastax.oss.driver.api.core.session.Request;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.datastax.oss.driver.api.core.servererrors.CoordinatorException;
+import com.datastax.oss.driver.api.core.servererrors.OverloadedException;
+import com.datastax.oss.driver.api.core.servererrors.WriteFailureException;
+import com.datastax.oss.driver.api.core.retry.RetryDecision;
+import com.datastax.oss.driver.api.core.retry.RetryPolicy;
+
+import java.util.Random;
+
+/**
+ * Implements a Cassandra {@link RetryPolicy} with back-offs for
+ * {@link OverloadedException} failures
+ * <p>
+ * {@link #maxRetryCount} specifies the number of retries that should be
+ * attempted. A value of -1 specifies that an indefinite number of retries
+ * should be attempted. For {@link #onReadTimeout}, {@link #onWriteTimeout}, and
+ * {@link #onUnavailable}, we retry immediately. For onRequestErrors such as
+ * OverLoadedError, we try to parse the exception message and use RetryAfterMs
+ * field provided from the server as the back-off duration. If RetryAfterMs is
+ * not available, we default to exponential growing back-off scheme. In this
+ * case the time between retries is increased by
+ * {@link #growingBackOffTimeMillis} milliseconds (default: 1000 ms) on each
+ * retry, unless maxRetryCount is -1, in which case we back-off with fixed
+ * {@link #fixedBackOffTimeMillis} duration.
+ * </p>
+ */
+public final class CosmosRetryPolicy implements RetryPolicy {
```

Comment:
Is this referenced anywhere in a sample?

Reply:
Yes. See readme.md and reference.conf. Custom retry policy is activated from config file.